task2/rest-api-basics/repository/src/main/java/com/epam/esm/repository/GiftCertificateRepository.java

```diff
@@ -0,0 +1,26 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.repository.exception.RepositoryException;
+import org.springframework.data.domain.Sort;
+
+import java.util.List;
+import java.util.Set;
+
+public interface GiftCertificateRepository extends Repository<GiftCertificate, Long> {
+    GiftCertificate findByName(String name) throws RepositoryException;
+    List<GiftCertificate> findByTagName(String tagName) throws RepositoryException;
+    Long updateById(Long id, GiftCertificate giftCertificate) throws RepositoryException;
+    Long deleteByName(String name) throws RepositoryException;
+
+    Set<GiftCertificate> findAllWithTags() throws RepositoryException;
```

Comment:
@igor-zhuravlev Why do you need so many identical methods? Can you define the execution logic at the service level, generate a request, and call one method? There are a lot of unnecessary methods.